### PR TITLE
Refactoring Tic Tac Toe with Redux

### PR DESCRIPTION
**Pull request by Dorian30**
## Summary
- Refactored Tic Tac Toe to Redux structure. 
- Set a game reducer to handle changes made to the state of the Tic Tac Toe. 
- Defined several actions to control the events of the app.

## Screenshot
redux

## Trello Card
https://trello.com/c/bm4y3a69/14-redux-i-actions-reducers-store-todo-finger

### CODE REVIEW

File: react-app/src/redux/game/reducer.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        squares
+      }
+    ]),
+    xIsNext: !state.xIsNext,
+    stepNumber: history.length,
+    winner
+  };
+};
+
+const jumpTo = (state, step) => {
```
Comment (LucasZibell) [inline react-app/src/redux/game/reducer.js]: same

File: react-app/src/redux/game/reducer.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      squares: Array(9).fill(null)
+    }
+  ],
+  stepNumber: 0,
+  xIsNext: true,
+  winner: null
+};
+
+const handleClick = (state, squarePosition) => {
```
Comment (LucasZibell) [inline react-app/src/redux/game/reducer.js]: Try to have all this logic in the action, the reducer should be to just update the state

File: react-app/src/redux/game/reducer.js (hunk 1 above)
Comment (LucasZibell) [reply react-app/src/redux/game/reducer.js]: Also, this functions are pretty big, how about moving them to gameActionUtils.js and export them from there

File: react-app/src/redux/game/actions.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const actions = {
+  HANDLE_CLICK: 'HANDLE_CLICK',
+  JUMP_TO: 'JUMP_TO'
+};
+
+const actionCreators = {
+  handleClick: squarePosition => ({
+    type: actions.HANDLE_CLICK,
```
Comment (LucasZibell) [inline react-app/src/redux/game/actions.js]: How about a better name? Like play turn or something like that. Keep in mind in the action you will have tons of actions, an `handleClick` is pretty generic

File: react-app/src/redux/game/actions.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    squarePosition
+  }),
+  jumpTo: step => ({
+    type: actions.JUMP_TO,
```
Comment (LucasZibell) [inline react-app/src/redux/game/actions.js]: same

**Review by LucasZibell — CHANGES_REQUESTED** [review]

**Review by HarryPerez — APPROVED** [review]

**Review by LucasZibell — APPROVED** [review]

**Pull request merged**